### PR TITLE
New file div

### PR DESCRIPTION
**Pull request by jdalzatec**
image

### CODE REVIEW

File: src/components/upload/FileDiv.tsx [hunk 1]
```diff
@@ -59,12 +95,86 @@ interface Props {
...
+    const identifier = "PT ";
+    return text.split("\n").filter((line) => line.startsWith(identifier))
+      .length;
+  };
+
+  const countReferences = (text: string) => {
+    const identifier = "NR ";
+    return text
+      .split("\n")
+      .filter((line) => line.startsWith(identifier))
```
Comment (dsvalenciah) [inline src/components/upload/FileDiv.tsx:149]: I saw that you're using arrow functions with one parameter with parenthesis, in this specific cases, can be without parenthesis, we can start using prettier and configure it to automatically prettify on save in our editors, make sense?

File: src/components/upload/FileDiv.tsx [hunk 2]
```diff
@@ -59,12 +95,86 @@ interface Props {
...
+      .map((line) => parseInt(line.replace(identifier, "")))
+      .reduce((n, m) => n + m);
+  };
+
+  useEffect(() => {
+    fileBlob.text().then((text) => {
```
Comment (dsvalenciah) [inline src/components/upload/FileDiv.tsx:155]: Upload a file and analyze it on the front-end can be a really big task and can consume a lot of memory if the file is to big, so, I think that we can start using a limit of it size, and in the future we can move the logic of analyze the file to our backend, or what do you think?

**Review by dsvalenciah — APPROVED** [review]
I think that it was a really great job, actually I saw a lot of things to learn of css, tan bello.

File: src/components/upload/FileDiv.tsx (hunk 1 above)
Comment (jdalzatec) [reply src/components/upload/FileDiv.tsx:149]: yes, my friend. Prettier (our formatter) wraps up the parameter in parenthesis, but we could remove them.

Comment (odarbelaeze) [reply src/components/upload/FileDiv.tsx:149]: This is the default prettier config.

File: src/components/upload/FileDiv.tsx (hunk 2 above)
Comment (odarbelaeze) [reply src/components/upload/FileDiv.tsx:155]: This processing is like nothing compared to the memory footprint of react, we might limit the number of files an user can upload tho.

**Pull request merged**